src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -289,12 +289,13 @@ class DependencyGraphContext:
     session_state: CurrentSessionState
 
 
-class Dependency(abc.ABC):
+class Dependency:
 
-    def __init__(self, loader: DependencyLoader, path: Path, inherits_context=True):
+    def __init__(self, loader: DependencyLoader, path: Path, inherits_context=True, lineage_str: str | None = None):
```

Comment:
```suggestion
    def __init__(self, loader: DependencyLoader, path: Path, inherits_context=True, lineage_str: list[Link] | None = None):
```

```
@dataclass
class Link:
    object_type: str
    object_id: str
```

let's use typed entries and not just strings. selialize to JSON on storage. If you want - do it in a separate PR.

Reply:
Created https://github.com/databrickslabs/ucx/issues/2550

Reply:
can you address it in this PR or decouple lineage from it?

Reply:
I can merge the PR but maybe you want to review it first ?

Reply:
See https://github.com/databrickslabs/ucx/pull/2556